### ISSUE TITLE
Bounty #1 for Homayoun

### ISSUE THREAD

**Issue by kirillt** [ARK-Builders/arklib-android#22]
We need to solve several issues at once:

1) https://github.com/ARK-Builders/arklib-android/issues/20
Ensure that everywhere ResourceID is taken from arklib, right now it is CRC32 checksum + filesize.
When we need to print a string with both values, we can just delimit them using dash (`-`).
     - [ ] Update ARK-Shelf
     - [ ] Update ARK-Shelf-Desktop
     - [ ] Update ARK-Navigator
           We should also implement migration in ARK Navigator, i.e. check some value (or index version) in app data and if there is none (or version is too low) then perform storages upgrade. Index and previews could be just dropped. Storages must be backed up before performing upgrade and old resource ids must be replaced by new resource ids. E.g. `5839051: english, greek` must be replaced with `88363-5839051: english, greek` in `.ark/tags`, etc.

2) https://github.com/ARK-Builders/arklib-android/issues/21
Basically, we just need to take code from Navigator and move it into arklib-android, ensuring any app could just import couple of modules and use metadata as well as index. We don't really need indexing in the apps right now (except Navigator). Could be useful later.

3) https://github.com/ARK-Builders/ARK-Memo/issues/8
We need to use new metadata storage and separate `title` into it. Also, we can introduce `created-date` metadata field.
This issue is kinda similar to https://github.com/ARK-Builders/ARK-Shelf/pull/21

**Comment by kirillt** [comment]
Just noticed that Navigator writes empty metadata storages to disk:
```
[kirill@lenovo ~]$ for f in Pictures/.ark/meta/* ; do cat $f ; echo ; done
null
null
null
null
null
null
```

I think, we should not write empty storages. Not all resource kinds use metadata storage at all, so better to keep disk clean.

**Comment by HLABZcc** [comment]
Interested! 😎

**Comment by kirillt** [comment]
@hhio618 btw I haven't seen you applied for the bounty itself..
https://gitcoin.co/issue/29473

**Comment by ginanisque** [comment]
Hi. I would like to know if this bounty is still available?

**Comment by kirillt** [comment]
This set of tasks was split into smaller bounties.
